pkg/ingester/shipper.go

```diff
@@ -286,18 +317,71 @@ func writeShipperMetaFile(logger log.Logger, dir string, meta *shipperMeta) erro
 	if err := f.Close(); err != nil {
 		return err
 	}
+
+	// Keep backwards compatibility with the previous Mimir version.
```

Comment:
This is fine, but why not finish writing mimir-file first (by doing the rename)?